src/prefect/flows.py

```diff
@@ -158,6 +159,25 @@ def __init__(
             List[Callable[[schemas.core.Flow, FlowRun, State], None]]
         ] = None,
     ):
+        # Validate if hooks type is list and hook_type is callable
+        hooks = [on_completion, on_failure, on_cancellation, on_crashed]
+        for hook_types in hooks:
+            if hook_types is not None:
+                try:
+                    hook_types = list(hook_types)
+                except TypeError:
+                    raise TypeError(
+                        f"Expected list of callable for {hook_types}; got"
```

Comment:
This could probably say "Expected iterable for..." since we're covering callables in the next message

Reply:
We'll want to make sure this also points the user to the correct parameter i.e. "Expected iterable for 'on_cancellation'; got..." — right now it looks like it will display the value they passed.